src/std/file.c

```diff
@@ -44,6 +47,15 @@ HL_PRIM hl_fdesc *hl_file_open( vbyte *name, int mode, bool binary ) {
 #	ifdef HL_UFOPEN
 	static const uchar *MODES[] = { USTR("r"), USTR("w"), USTR("a"), NULL, USTR("rb"), USTR("wb"), USTR("ab") };
 	FILE *f = fopen((uchar*)name,MODES[mode|(binary?4:0)]);
+#   elif defined(HL_IOS) || defined(HL_TVOS) || defined(HL_ANDROID)
+    static const char *MODES[] = { "r", "w", "a", NULL, "rb", "wb", "ab" };
+    int m = mode | (binary?4:0);
+    FILE *f = NULL;
+    if (m==1 || m>=5){
+        f = fopen((char*)hl_mobile_get_document_path(name),MODES[m]);
+    }else{
+        f = fopen((char*)hl_mobile_get_resource_path(name),MODES[m]);
+    }

```

Comment:
The prefixing of directories with a specific path should be done at Haxe level, this shouldn't be much a problem as Heaps as very few file reads